### PR TITLE
fix: allow more than 65535 repos in deny list

### PR DESCRIPTION
**Pull request by bahrmichael**
Closes https://github.com/sourcegraph/sourcegraph/issues/61353

This PR fixes an issue where too many user_repo_permissions would lead to an SQL error. We fix this by not loading all denied repo ids through the application layer back into the database, and instead loading the denied repo ids within the `WHERE` statement of the insights query.

TODO: 
- handle other call to GetUnauthorizedRepoIDsQuery
- create new ticket for early abort when we have more than 65535 predicates
- measure performance impact of avoiding roundtrip

## Test plan

- We didn't touch how permissions work, so this should be Covered by existing tests
- Added additional tests for the code that uses the resulting query

### CODE REVIEW

**Comment by bahrmichael** [comment]
There are currently a lot of compilation issues, mostly from other packages. I'll focus on ruling out problems from the authz variations before tackling the inter-package refactoring.

**Comment by bahrmichael** [comment]
Making sure that we don't negatively impact security may take a bit because there are nearly 1k lines in `repos_perm_test.go` which tests for correct auth.

File: internal/database/dbtest/dbtest.go [hunk 1]
```diff
@@ -93,7 +93,7 @@ func getSchemaAtRev(t testing.TB, name, rev string) *schemas.Schema {
 // NewInsightsDB returns a connection to a clean, new temporary testing database with
 // the same schema as Sourcegraph's CodeInsights production Postgres database.
 func NewInsightsDB(logger log.Logger, t testing.TB) *sql.DB {
-	return newDB(logger, t, "insights", schemas.CodeInsights)
+	return newDB(logger, t, "insights", schemas.CodeInsights, schemas.Frontend)
```
Comment (bahrmichael) [inline internal/database/dbtest/dbtest.go:96]: Todo for myself: Test if we can revert this change again.

Comment (bahrmichael) [reply internal/database/dbtest/dbtest.go:96]: Update: We need this line so that the authz query can run (it access the repo and user_repo_permissions tables).

Comment (camdencheek) [reply internal/database/dbtest/dbtest.go:96]: OH. I suddenly understand why we're doing it the way we're doing it now.

The Code Insights DB and the Frontend DB are two separate databases. A query cannot hit both of them. I'm pretty sure now this is why we queried the Frontend DB first, then used the results against the Code Insights DB.

So this means my comments about just using the query conditons directly were wrong. Sorry 😬

...so back to your ideas. Pagination might be tricky because this is a negative query. Thoughts?

Comment (bahrmichael) [reply internal/database/dbtest/dbtest.go:96]: Couple ideas come to mind:
- Documenting a service limitation that customers with more than 65k restricted repos cannot run code insights queries
- Introducing a different auth and role mechanism to enable this with less predicates (sounds like a lot of work to me, including migration of stuff)
- Introducing a replication of permissions from the frontend to the code insights db
- Instead of sending predicates to the code insights db, filter out restricted repos in the application layer (sounds like the most pragmatic approach, will work on that today)

Comment (bahrmichael) [reply internal/database/dbtest/dbtest.go:96]: I've spun up a new PR with the last idea: https://github.com/sourcegraph/sourcegraph/pull/61580
